udica-templates/base_container.cil

```diff
@@ -0,0 +1,94 @@
+;;
+;; Macros
+;;
+
+(macro search_dirs ((type ARG1) (type ARG2))
+	(allow ARG1 ARG2 (dir (getattr search open)))
+)
```

Comment:
search_dirs does not need "open" (i was reminded of that by @cgzones)

Reply:
Also i would have used classpermissionset (template from the bottom up)

```
(classpermission·search_dir)$
(classpermissionset·search_dir·(dir·(getattr·search)))

(macro search_dirs ((type ARG1)(type ARG2))
            (allow ARG1 ARG2 search_dir))
```

Reply:
Thanks. I opted to replace the macro calls with classpermissionsets instead as I think they are more clear.

Also removed the `open` permission from `search_dir_perms`.